datafusion/core/src/physical_optimizer/join_selection.rs

```diff
@@ -157,7 +157,9 @@ fn swap_join_projection(
 }
 
 /// This function swaps the inputs of the given join operator.
-fn swap_hash_join(
+/// This function is public so other downstream projects can use it
```

Comment:
```suggestion
```

Reply:
Without a clear comment, it might be wrongly moved out of public APIs easily.

Reply:
I agree but the entire idea of `pub` modifier implies 3rd party users to use the pub method? 🤔